validator_client/src/block_service.rs

```diff
@@ -357,59 +367,212 @@ impl<T: SlotClock + 'static, E: EthSpec> BlockService<T, E> {
+                                        );
+                                    }
                                 }
-                            }
-                            Err(BlockError::Irrecoverable(e)) => {
-                                // Only log an `error` since it's common for
-                                // builders to timeout on their response, only
-                                // to publish the block successfully themselves.
-                                error!(
+                                Err(BlockError::Irrecoverable(e)) => {
+                                    // Only log an `error` since it's common for
+                                    // builders to timeout on their response, only
+                                    // to publish the block successfully themselves.
+                                    error!(
+                                        log,
+                                        "Error whilst producing block";
+                                        "error" => ?e,
+                                        "block_slot" => ?slot,
+                                        "info" => "this error may or may not result in a missed block",
+                                    )
+                                }
+                                Ok(_) => {}
+                            };
+                        } else if let Err(e) = service
+                            .publish_block::<FullPayload<E>>(slot, validator_pubkey)
+                            .await
+                            {
+                                // Log a `crit` since a full block (non-builder)
+                                // proposal failed.
+                                crit!(
                                     log,
                                     "Error whilst producing block";
-                                    "error" => ?e,
+                                    "message" => ?e,
                                     "block_slot" => ?slot,
-                                    "info" => "this error may or may not result in a missed block",
-                                )
+                                    "info" => "proposal did not use a builder",
+                                );
                             }
-                            Ok(_) => {}
-                        };
-                    } else if let Err(e) = service
-                        .publish_block::<FullPayload<E>>(slot, validator_pubkey)
-                        .await
-                    {
-                        // Log a `crit` since a full block (non-builder)
-                        // proposal failed.
-                        crit!(
+                    },
+                    "block service",
+                )
+            }
+        }
+
+        Ok(())
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    async fn handle_block_response<Payload: AbstractExecPayload<E>>(
+        &self,
+        log: &Logger,
```

Comment:
nit: we could probably use `self.context.log()` instead to reduce the number of arguments here